ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -59,121 +62,119 @@ public CCodeGenerator() {
-        final Path outputPath = getRequestOutputPath(request);
+            if (StructHelper.isPrimitive(structEntry) || StructHelper.containsExistingStructs(structEntry, existingTypesBuilder.build())) {
+                existingTypesBuilder.add(StructHelper.getResponseTypeName(structEntry.getName()));
+                orderedStructsBuilder.add(allStructs.remove());
+            } else {  // move to end to come back to
+                allStructs.add(allStructs.remove());
+            }
 
-        final OutputStream outStream = fileUtils.getOutputFile(outputPath);
-        final Writer writer = new OutputStreamWriter(outStream);
-        try {
-            requestTemplate.process(request, writer);
-        } catch (final NullPointerException e) {
-            LOG.error("Encountered NullPointerException while processing template " + requestTemplate.getName(), e);
+            if (allStructsSize == allStructs.size()) {
+                skippedStructsCount++;
+                if (skippedStructsCount == allStructsSize) {
+                    LOG.warn("Unable to progress on remaining structs, aborting!");
+                    LOG.warn("  Remaining structs[" + allStructs.size() + "]");
+                    for (final Struct struct : allStructs) {
+                        LOG.warn("    " + struct.getName());
+                    }
+                    break;
+                }
+            }
         }
+
+        return orderedStructsBuilder.build();
     }
 
-    public void generateTypeTemplate(final Type typeEntry, final String templateName) throws IOException {
-        final Template typeTemplate = config.getTemplate(templateName);
+    public static ImmutableList<Request> getAllRequests(Ds3ApiSpec spec) throws ParseException {
+        final ImmutableList.Builder<Request> allRequestsBuilder = ImmutableList.builder();
+        if (ConverterUtil.hasContent(spec.getRequests())) {
+            for (final Ds3Request ds3Request: spec.getRequests()) {
+                allRequestsBuilder.add(RequestConverter.toRequest(ds3Request));
+            }
+        }
+        return allRequestsBuilder.build();
+    }
 
-        final Path outputPath = getTypeOutputPath(typeEntry);
+    public void processTemplate(final Object obj, final String templateName, final OutputStream outputStream) throws IOException {
+        final Template template = config.getTemplate(templateName);
 
-        final OutputStream outStream = fileUtils.getOutputFile(outputPath);
-        final Writer writer = new OutputStreamWriter(outStream);
+        final Writer writer = new OutputStreamWriter(outputStream);
         try {
-            typeTemplate.process(typeEntry, writer);
+            template.process(obj, writer);
         } catch (final NullPointerException e) {
-            LOG.error("Encountered NullPointerException while processing template " + typeTemplate.getName(), e);
+            LOG.error("Encountered NullPointerException while processing template " + templateName, e);
```

Comment:
It looks like these two exceptions can be combined into one.  I don't believe that you need to call out the specific exception since it will print the stack trace for the exception.